app/scripts/metamask-controller.js

```diff
@@ -4454,6 +4456,38 @@ export default class MetamaskController extends EventEmitter {
 
       await this._addAccountsWithBalance(id);
 
+      const client = new KeyringInternalSnapClient({
+        messenger: this.controllerMessenger,
+        snapId: SOLANA_WALLET_SNAP_ID,
+      });
+      const keyring = await this.getSnapKeyring();
+      for (let index = 0; ; index++) {
+        const discovered = await client.discoverAccounts(
+          [SolScope.Mainnet, SolScope.Testnet, SolScope.Devnet],
+          id,
+          index,
+        );
+
+        if (discovered.length === 0) {
+          break;
+        }
+
+        await Promise.all(
+          discovered.map(async (discoveredAccount) => {
+            const options = {
+              scope: discoveredAccount.scopes[0],
```

Comment:
I'd add a `FIXME` here, cause if Solana returns multiple `scopes` (they should be, since 1 Solana account is compatible with multiple Solana networks), so something like:

```suggestion
              // FIXME: We should pass all scopes here.
              scope: discoveredAccount.scopes[0],
```